core/dotnet/src/AzureIdentityAuthenticationProvider.cs

```diff
@@ -0,0 +1,31 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Azure.Core;
+using Kiota.Abstractions;
+
+namespace KiotaCore {
+    public class AzureIdentityAuthenticationProvider : IAuthenticationProvider
+    {
+        private readonly TokenCredential creds;
+        private readonly List<string> _scopes;
+        public AzureIdentityAuthenticationProvider(TokenCredential credentials, params string[] scopes)
+        {
+            creds = credentials ?? throw new ArgumentNullException(nameof(credentials));
+            if(_scopes == null)
+                _scopes = new();
+            else
+                _scopes = scopes.ToList();
+
+            if(!_scopes.Any())
+                _scopes.Add("https://graph.microsoft.com/.default");
```

Comment:
This can be problematic when targeting national/air-gapped clouds. We should add a TODO for this or use the hostname from `requestUri` in `getAuthorizationToken(Uri requestUri)` to build the default scope url.

Reply:
added a todo